engine/src/test/java/org/camunda/bpm/engine/test/api/context/DelegateExecutionContextTest.java

```diff
@@ -42,70 +47,197 @@
  */
 public class DelegateExecutionContextTest {
 
-  protected static final BpmnModelInstance DELEGATION_PROCESS = Bpmn.createExecutableProcess("process1")
+  protected static final BpmnModelInstance SERVICE_TASK_DELEGATE_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+      .serviceTask("serviceTask1")
+        .camundaClass(DelegateClass.class.getName())
+      .endEvent()
+      .done();
+
+
+  protected static final BpmnModelInstance EXEUCTION_LISTENER_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+        .camundaExecutionListenerClass(ExecutionListener.EVENTNAME_START, ExecutionListenerImpl.class.getName())
+      .endEvent()
+      .done();
+
+  protected static final BpmnModelInstance SIGNAL_EVENT_PROCESS = Bpmn.createExecutableProcess()
+      .startEvent()
+        .intermediateCatchEvent("catchSignal")
+          .signal("${delegateExecutionContextBean.getCurrentActivityId()}")
+      .endEvent()
+      .done();
+
+  protected static final BpmnModelInstance ONE_TASK_PROCESS_SUBPROCESS_SIGNAL_INTERMEDIATE_EVENT = Bpmn.createExecutableProcess()
+      .startEvent()
+        .userTask("A")
+      .endEvent()
+      .subProcess("subprocess")
```

Comment:
This creates semantically invalid BPMN, because it creates an outgoing flow from the end event to the subprocess.